### ISSUE TITLE
window extent error associated with formation/group labeling

### ISSUE THREAD

**Issue by Swanson-Hysell** [yuempark/pystrat#13]
I am currently getting this error with the first *Quick Start* cell of the `pystrat-tutorial.ipynb`:

```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
/var/folders/57/lmgz7h4x11v34k53c6ldrlrw0000gn/T/ipykernel_5592/926299459.py in <module>
     20 # ylim to accommodate the section
     21 ax = plt.axes(ylim=[0, 120])
---> 22 section_01.plot(style_01, ax=ax, label_units=True, unit_label_wid_tot=0.3)
     23 plt.show()

~/0000_GitHub/pystrat/pystrat/pystrat.py in plot(self, style, ax, linewidth, annotation_height, label_units, unit_label_wid_tot, unit_fontsize)
    736                     # get resulting data coordinates of the plotted label
    737                     transf = ax.transData.inverted()
--> 738                     bb = cur_text.get_window_extent()
    739                     bb_datacoords = bb.transformed(transf)
    740                     # if overstepping the box, give a warning and remove

~/opt/anaconda3/lib/python3.9/site-packages/matplotlib/text.py in get_window_extent(self, renderer, dpi)
    905             self._renderer = self.figure._cachedRenderer
    906         if self._renderer is None:
--> 907             raise RuntimeError('Cannot get window extent w/o renderer')
    908 
    909         with cbook._setattr_cm(self.figure, dpi=dpi):

RuntimeError: Cannot get window extent w/o renderer
```

I think that a fix could be to put in code such as `plt.gcf().canvas.draw()` that forces the figure to be rendered prior to `cur_text.get_window_extent()`

**Comment by sarttiso** [comment]
From a cursory search, this seems likely to be a mac osx issue. What is the output of `matplotlib.backend()` for you?

If you run `matplotlib.use('tkagg')` does it work then (it should create a figure in a new window)?

If you add a code cell with `%matplotlib inline` before the first quick start cell, does it work then?

**Comment by Swanson-Hysell** [comment]
`print(matplotlib.get_backend())` gives `module://matplotlib_inline.backend_inline`

Behavior doesn't change with `%matplotlib inline` addition (given that this is already the backend).

Still get error with `matplotlib.use('tkagg')` (see below). I think that pull request #14 could be the solution to fix the issue. If you run things in that branch (https://github.com/yuempark/pystrat/tree/get_window_extent-fix) does it work for you? 

```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
/var/folders/57/lmgz7h4x11v34k53c6ldrlrw0000gn/T/ipykernel_48814/926299459.py in <module>
     20 # ylim to accommodate the section
     21 ax = plt.axes(ylim=[0, 120])
---> 22 section_01.plot(style_01, ax=ax, label_units=True, unit_label_wid_tot=0.3)
     23 plt.show()

~/0000_GitHub/pystrat/pystrat/pystrat.py in plot(self, style, ax, linewidth, annotation_height, label_units, unit_label_wid_tot, unit_fontsize)
    736                     # get resulting data coordinates of the plotted label
    737                     transf = ax.transData.inverted()
--> 738                     bb = cur_text.get_window_extent()
    739                     bb_datacoords = bb.transformed(transf)
    740                     # if overstepping the box, give a warning and remove

~/opt/anaconda3/lib/python3.9/site-packages/matplotlib/text.py in get_window_extent(self, renderer, dpi)
    905             self._renderer = self.figure._cachedRenderer
    906         if self._renderer is None:
--> 907             raise RuntimeError('Cannot get window extent w/o renderer')
    908 
    909         with cbook._setattr_cm(self.figure, dpi=dpi):

RuntimeError: Cannot get window extent w/o renderer
```

**Comment by sarttiso** [comment]
Huh, these OS-specific issues are a real pain. Your fix did work on my end, and I merged #14 .

**Comment by Swanson-Hysell** [comment]
Glad that fix worked on your end too!
